packages/workspace/src/tasks-runner/cache.ts

```diff
@@ -164,6 +163,66 @@ export class Cache {
     }
   }
 
+  removeTaskLocalOutputsHash(task: Task, outputs: string[]): void {
+    if (outputs.length === 0) {
+      return;
+    }
+
+    const hashFile = this.getTaskLocalHashFile(task);
+    try {
+      unlinkSync(hashFile);
+    } catch (e) {}
+  }
+
+  writeTaskLocalOutputsHash(task: Task, outputs: string[]): void {
+    if (outputs.length === 0) {
+      return;
+    }
+
+    const hashFile = this.getTaskLocalHashFile(task);
+    writeToFile(hashFile, task.hash);
+  }
+
+  localOutputsHashMatchTaskHash(
```

Comment:
I find this is hard ot read. 

Something like this can be better:

  shouldCopyFileFromCache(
    taskWithCacheResult: TaskWithCachedResult,
    outputs: string[]
  ): boolean {
    if (getLatestRecordedHashForTask(taskWithCacheResult.task) !== taskWithCacheResult.task.hash) return true;
    return !allOutputsExists(outputs) 
  }
  
  getLatestRecordedHashForTask(task: Task) {
    try {
      return readFileSync(this.getLatestRecordedHashForTask(task)).toString();
    } catch (e) {
        return nul;
    }
   }